Nos/Service/Relay/RelayService.swift

```diff
@@ -630,14 +630,21 @@ extension RelayService {
         }
     }
 
-    func relayAddresses(for user: CurrentUser) async -> [URL] {
-        await backgroundContext.perform { () -> [URL] in
+    func relayAddresses(for user: CurrentUser) async -> Set<URL> {
+        await backgroundContext.perform { () -> Set<URL> in
             if let currentUserPubKey = user.publicKeyHex,
                 let currentUser = try? Author.find(by: currentUserPubKey, context: self.backgroundContext) {
-                let userRelays = currentUser.relays
-                return userRelays.compactMap { $0.addressURL }
+                let userRelays = Set(currentUser.relays.compactMap { $0.addressURL })
+                
+                // Remove search only relays
+                let filteredRelays = userRelays.filter { relayAddress in
+                    guard let host = relayAddress.host(percentEncoded: true) else { return true }
+                    return !host.hasSuffix(".nostr.band")
```

Comment:
Is there a good way to use the `searchOnly` array you defined? I realize we don't want an exact match, so maybe there's not, but I want to at least think about it.

Reply:
I'm not seeing it, and it's probably not worth spending more time thinking about it, so I'm good with leaving this as-is.